File: eBarberShop/eBarberShop.Model/SlikeUsluge.cs
namespace eBarberShop.Model
{
    public class SlikeUsluge
    {
        public int SlikeUslugeId { get; set; }
        public int SlikaId { get; set; }

        public virtual Slike Slika { get; set; }

        public byte[]? SlikaUsluga => Slika?.Slika;
        public string? OpisSlike => Slika?.Opis;
        public DateTime? DatumObjave => Slika?.DatumPostavljanja;
    }
}

File: eBarberShop/UI/ebarbershop_mobile/lib/models/slike_usluge/slike_usluge.dart
import 'package:json_annotation/json_annotation.dart';

part 'slike_usluge.g.dart';

@JsonSerializable()
class SlikeUsluge {
  int slikeUslugeId;
  int slikaId;
  String? slikaUsluga;
  String? opisSlike;
  DateTime? datumObjave;

  SlikeUsluge(this.slikeUslugeId, this.slikaId, this.slikaUsluga, this.opisSlike, this.datumObjave);

  factory SlikeUsluge.fromJson(Map<String, dynamic> json) =>
      _$SlikeUslugeFromJson(json);

  Map<String, dynamic> toJson() => _$SlikeUslugeToJson(this);
}